### PR TITLE
use srcset

### PR DESCRIPTION
**Pull request by justame**

### CODE REVIEW

**Comment by justame** [comment]
@lxgreen, please review

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 1]
```diff
@@ -203,11 +222,16 @@ class ImageViewer extends React.Component {
 
   getImage(imageClassNames, src, alt, props, opts = {}) {
     const { fadeIn = false, width, height } = opts;
+    let srcSet;
+    if (this.shouldUseSrcSet()) {
+      srcSet = replaceUrlFileExtenstion(src, '.webp');
+    }
```
Comment (oded-soffrin) [inline packages/plugin-image/web/src/image-viewer.jsx:228]: didn't we said we need it only for preloaded image?

Comment (justame) [reply packages/plugin-image/web/src/image-viewer.jsx:228]: I don't see a problem doing it for both

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 2]
```diff
@@ -203,11 +222,16 @@ class ImageViewer extends React.Component {
...
     return (
       <img
         {...props}
         className={imageClassNames}
         src={src}
+        srcSet={srcSet}
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:234]: check with media when webp is supported

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 3]
```diff
@@ -22,6 +22,20 @@ import InPluginInput from './InPluginInput';
 
 const isSafari = () => /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
 
+const replaceUrlFileExtenstion = (url, extensionTarget) => {
+  const originalExtensionFound = url.lastIndexOf(extensionTarget) > -1;
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:26]: take the last 4/5 chars.

**Comment by oded-soffrin** [comment]
It’s under experiment, so why not. I have two open questions before opening:
- will there always be a webp format available?
- the highres is client only, where we used the imageClientAPI. if it doesn’t return webp, why would we want to force replace? it’s probably not supported, so I think it’s safer to use only in preload, and the switch should happen [here](https://github.com/wix/ricos/blob/master/packages/ricos-content/web/src/imageUtils.ts#L67)  where we generate the URL

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 1 above)
Comment (oded-soffrin) [reply packages/plugin-image/web/src/image-viewer.jsx:228]: see comment below

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 3 above)
Comment (lxgreen) [inline packages/plugin-image/web/src/image-viewer.jsx:26]:
```suggestion
return url.replace(/(.*)\.(jp(e)?g|png)$/, `$1.${extensionTarget}`)
```

**Review by lxgreen — COMMENTED** [review]
a shorter solution

**Comment by justame** [comment]
I talked with Media and they always support webp for images

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 2 above)
Comment (justame) [reply packages/plugin-image/web/src/image-viewer.jsx:234]: they support

File: packages/plugin-image/web/src/image-viewer.tsx [hunk 4]
```diff
@@ -75,6 +80,21 @@ class ImageViewer extends React.Component<ImageViewerProps, ImageViewerState> {
 
   static contextType = GlobalContext;
 
+  shouldSkipImageThumbnail = () => {
```
Comment (oded-soffrin) [inline packages/plugin-image/web/src/image-viewer.tsx:83]: did you merge from master? I've removed all the `shouldSkipImageThumbnail` experiment

Comment (justame) [reply packages/plugin-image/web/src/image-viewer.tsx:83]: I had a big conflict, I'll clean it

**Pull request merged**